src/requests/requests.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { UserEntity } from 'src/users/dto/UserEntity.dto'
+import { CreateRequestDto } from './dto/CreateRequest.dto'
+import { RequestDetailsDto } from './dto/RequestDetails.dto'
+import { RequestPreviewDto } from './dto/RequestPreview.dto'
+import { UpdateRequestDto } from './dto/UpdateRequest.dto'
+
+@Injectable()
+export class RequestsService {
+  constructor(private prisma: PrismaService) {}
+
+  async findAll(): Promise<RequestPreviewDto[]> {
+    const requests = await this.prisma.consultationRequest.findMany({
+      include: {
+        initializer: true,
+        subject: true,
+        _count: {
+          select: {
+            supporters: true,
+            consultation: true,
+          },
+        },
+      },
+    })
+
+    return requests.map(({ initializer, _count, ...request }) => {
+      return {
+        ...request,
+        initializer: { fullName: initializer.fullName, id: initializer.id },
```

Comment:
removing the sensitive info from the user should be done in the db query. Instead of` include: { initializer: true }`, you can do` include: { initializer: publicUserProjection} `like in other services